rustler/src/types/map.rs

```diff
@@ -182,3 +183,20 @@ impl<'a> Decoder<'a> for MapIterator<'a> {
         }
     }
 }
+
+impl<'a, T> Decoder<'a> for RangeInclusive<T>
+where
+    T: Decoder<'a>,
+{
+    fn decode(term: Term<'a>) -> NifResult<Self> {
+        let vec:Vec<(Term, Term)> = term.decode::<MapIterator>()?.collect();
+        match (&*vec[0].0.atom_to_string()?, &*vec[0].1.atom_to_string()?) {
```

Comment:
This strikes me as fairly fragile, since erlang maps are unordered, our __struct__ key doesn't necessarily end up here. Might be a better idea to use the `Term.map_get` function here?